diff --git a/node_modules/hexo-server/lib/middlewares/header.js b/node_modules/hexo-server/lib/middlewares/header.js
index f623846..501bb1a 100644
--- a/node_modules/hexo-server/lib/middlewares/header.js
+++ b/node_modules/hexo-server/lib/middlewares/header.js
@@ -6,6 +6,7 @@ module.exports = function(app) {
 
   app.use((req, res, next) => {
     res.setHeader('X-Powered-By', 'Hexo');
+    res.setHeader('Access-Control-Allow-Origin', '*');
     next();
   });
 };
